pkg/runner/executable_path_resolving_runner.go

```diff
@@ -0,0 +1,121 @@
+package runner
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+
+	"github.com/buildbarn/bb-remote-execution/pkg/proto/runner"
+	"github.com/buildbarn/bb-storage/pkg/filesystem"
+	"github.com/buildbarn/bb-storage/pkg/util"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type rootDirectoryOpener func(inputRootDirectory string) (filesystem.DirectoryCloser, error)
```

Comment:
Rule of thumb: don't let a public API depend on private types. For example, don't let public functions take private types or return them. This makes them less flexible.

By making it public, you could add code to internal/mock to automatically generate a mock for RootDirectoryOpener, thereby making the unit tests a bit cleaner.